src/main/java/com/icthh/xm/uaa/security/DomainTokenServices.java

```diff
@@ -189,8 +190,16 @@ public OAuth2AccessToken refreshAccessToken(String refreshTokenValue, TokenReque
             throw new InvalidGrantException("Invalid refresh token: " + refreshTokenValue);
         }
 
-        OAuth2RefreshToken refreshToken = tokenStore.readRefreshToken(refreshTokenValue);
+        OAuth2RefreshToken refreshToken;
+        try {
+            refreshToken = tokenStore.readRefreshToken(refreshTokenValue);
+        } catch (InvalidTokenException e) {
+            log.warn("Could not refresh token, invalid refresh token: {}", refreshTokenValue);
+            throw new InvalidGrantException("Invalid refresh token: " + refreshTokenValue);
```

Comment:
i think it worth adding the original exception as a cause. Otherwise, you will lose the real stack trace to the problem.

Reply:
indeed, agree